### DISCUSSION TITLE
[QUESTION] Building a Web API

### DISCUSSION THREAD

**Discussion by carlosbertoncelli** [milvus-io/milvus#6596] [Q&A and General discussion]
Hi Guys,

I'm building a Web API using PyMilvus to handle the connections, insertions, searchs and drops of the collections.

I have a few questions regarding to the implementation:

I've seen the example.py documentation, but for my case it's not clear to me if my webservice should have one connection with milvus and do all the tasks through it or should i open a connection, do the operations and close it before sending request answer

Should i call load_collection in every request that needs it and after doing the needed operations should i release it or should i load the collection just once per worker and keep it that way?

**Comment by yhmo** [comment]
Some users of v1.x use Flask/FastAPI to build webservice for Milvus. The PyMilvus has a connection pool, so you can use one instance to handle all connections. The PyMilvus 2.0 is similar.

For load_collection():
In v1.x, no need to always call load_collection(), this method load data into cache so that search performance is best. If your collection data is static, only call load_collection() one time is enough. But if your collection data is streaming(continually/frequently insert), each search request will automatically load the "dynamic/tail segment" into cache, search performance is not best, but most of time it is acceptable.
In v2.0, there are two kinds of segments: sealed and growing. When you call load_collection(), the sealed segments will be load into query node's cache. Streaming data is inserted into growing segment, it always in memory. So you only need to call load_collection() once.
